components/organisms/o-order-summary.vue

```diff
@@ -0,0 +1,258 @@
+            :special-price="getProductSpecialPrice(product)"
+            :stock="product.stock.qty"
+            :qty="product.qty"
+            @click:remove="removeProduct(product)"
+            @input="changeQuantity(product, $event)"
+          >
+            <template #configuration>
+              <div class="product__properties">
+                <SfProperty
+                  v-for="property in product.options"
+                  :key="property.label"
+                  :name="property.label"
+                  :value="property.value"
+                  class="product__property"
+                />
+              </div>
+            </template>
+            <template #actions>
+              <div>
+                <div class="product__action">
+                  {{ product.sku }}
+                </div>
+                <div class="product__action">
+                  {{ $t("Quantity") }}:
+                  <span class="product__qty">{{ product.qty }}</span>
+                </div>
+              </div>
+            </template>
+          </SfCollectedProduct>
+        </div>
+      </transition>
+    </div>
+    <MPriceSummary class="highlighted highlighted--total" />
+    <div class="highlighted promo-code">
+      <SfButton
+        class="promo-code__button"
+        @click="showPromoCode = !showPromoCode"
+      >
+        {{ showPromoCode ? "-" : "+" }} {{ $t("Discount code") }}
+      </SfButton>
+      <transition name="fade">
+        <div v-if="showPromoCode">
+          <SfInput
+            v-model="promoCode"
+            name="promoCode"
+            :label="$t('Add a discount code')"
+            class="promo-code__input"
+            @keyup.enter="applyCoupon"
+          />
+          <SfButton class="sf-button--full-width" @click="applyCoupon">
+            {{ $t("Add discount code") }}
+          </SfButton>
+        </div>
+      </transition>
+    </div>
+    <div class="highlighted">
+      <SfCharacteristic
+        v-for="characteristic in characteristics"
+        :key="characteristic.title"
+        :title="characteristic.title"
+        :description="characteristic.description"
+        :icon="characteristic.icon"
+        class="characteristic"
+      />
+    </div>
+  </div>
+</template>
+<script>
+import { mapGetters } from 'vuex';
+import i18n from '@vue-storefront/i18n';
+import { getThumbnailForProduct } from '@vue-storefront/core/modules/cart/helpers';
+import {
+  SfInput,
+  SfButton,
+  SfHeading,
+  SfProperty,
+  SfCharacteristic,
+  SfCollectedProduct
+} from '@storefront-ui/vue';
+import MPriceSummary from 'theme/components/molecules/m-price-summary';
+export default {
+  name: 'OOrderSummary',
+  components: {
+    SfInput,
+    SfButton,
+    SfHeading,
+    SfProperty,
+    SfCharacteristic,
+    SfCollectedProduct,
+    MPriceSummary
+  },
+  data () {
+    return {
+      promoCode: '',
+      showPromoCode: false,
+      showProducts: false,
+      characteristics: [
```

Comment:
can we make it as props and this can be default value?

Reply:
i think if we can push business logic higher (in this example to page component) then we should try to do this (of course within reason)

Reply:
Done.